src/net/sourceforge/kolmafia/KoLmafia.java

```diff
@@ -192,12 +192,23 @@ public static String imageServerPath() {
   }
 
   public static boolean acquireFileLock(final String suffix) {
+    if (KoLmafia.SESSION_HOLDER != null) {
+      // If we have a session file lock already, release it.
+      KoLmafia.releaseFileLock();
+    }
     try {
       KoLmafia.SESSION_FILE = new File(KoLConstants.SESSIONS_LOCATION, "active_session." + suffix);
 
       if (KoLmafia.SESSION_FILE.exists()) {
         KoLmafia.SESSION_CHANNEL = new RandomAccessFile(KoLmafia.SESSION_FILE, "rw").getChannel();
         KoLmafia.SESSION_HOLDER = KoLmafia.SESSION_CHANNEL.tryLock();
+        if (KoLmafia.SESSION_HOLDER == null) {
+          KoLmafia.updateDisplay(
+              MafiaState.ABORT,
+              "Could not acquire file lock for "
+                  + suffix
+                  + ". Most likely, another instance of KoLmafia is logged in.");
```

Comment:
This is not wrapping in my login window and cutting off part of the text.  We could make it two lines, or more condensed.  Not a showstopper.